### PR TITLE
Preneeds swagger

### PR DESCRIPTION
**Pull request by cehsu**
## Background
create preneeds swagger documentation

#### Applies to all PRs

- [x] Appropriate test coverage & logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)

### CODE REVIEW

**Comment by lihanli** [comment]
can you add specs for this to `spec/request/swagger_spec.rb`?

File: spec/request/swagger_spec.rb [hunk 1]
```diff
@@ -191,14 +191,16 @@
...
     it 'supports adding a preneed claim' do
-      expect(subject).to validate(
-        :post,
-        '/v0/preneeds/burial_forms',
-        200,
-        '_data' => {
-          'application' => attributes_for(:burial_form)
-        }
-      )
+      VCR.use_cassette('preneeds/burial_forms/creates_a_pre_need_burial_form') do
```
Comment (cehsu) [inline spec/request/swagger_spec.rb]: Thanks for pointing this out @lihanli.

**Comment by jholton** [comment]
Sorry I'm leaving this very much a WIP as I go on vacation.
You can see from my changes that I'm in the middle of defining what a proper request looks like.
It's a bit annoying since what I'm basically doing is rewriting our schema in a Ruby DSL.
I could not find a way to just use our schema itself to create the definition.
That would also prevent this swagger file from getting stale if the schema changes.

I also noticed a few peculiarities.
The backend is accepting `applicationStatus` and `sendingCode`.  These are not in [our schema](https://github.com/department-of-veterans-affairs/vets-json-schema/blob/master/src/schemas/40-10007/schema.js).
However, they are in [this sample schema](https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Products/Pre-Needs-Burial/api/receive_preneeds_application_schema.json#L180).
Should references to them be removed from the backend?  Or should they be added to the schema/form?

**Comment by jholton** [comment]
Screenshot showing how this currently renders:

screen shot 2018-09-07 at 4 34 52 pm

**Comment by cehsu** [comment]
Thanks @jholton, an [issue](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/14289) has been created to investigate removing these potentially extraneous properties. Address property information has been added to the request. @lihanli @jholton PTAL.

File: app/swagger/requests/preneeds_claims.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                property :applicantRelationshipToClaimant, type: :string, example: 'Authorized Agent/Rep'
+                property :mailingAddress, type: :object do
+                  property :street, type: :string, example: '140 Rock Creek Church Rd NW'
+                  property :street2, type: :string, example: ''
+                  property :city, type: :string, example: 'Washington'
+                  property :country, type: :string, example: 'USA'
+                  property :state, type: :string, example: 'DC'
+                  property :postalCode, type: :string, example: '20011'
+                end
+              end
```
[the remark this answers is not in the document]
Comment (cehsu) [reply app/swagger/requests/preneeds_claims.rb]: Thanks @jholton, updated.

File: app/swagger/requests/preneeds_claims.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+          parameter do
+            key :name, :application
+            key :in, :body
+            key :description, 'Pre-need burial eligibility form data'
+            key :required, true
+
+            # TODO: add `required` designations
+
+            schema do
```
[the remark this answers is not in the document]
Comment (cehsu) [reply app/swagger/requests/preneeds_claims.rb]: Thanks @jholton, updated.

File: app/swagger/requests/preneeds_claims.rb [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            key :in, :body
+            key :description, 'Pre-need burial eligibility form data'
+            key :required, true
+
+            schema do
+              key :required, %i[applicant claimant hasCurrentlyBuried veteran]
+              property :applicationStatus, type: :string, example: 'example needed' # TODO: not in schema.  remove?
+              property :hasCurrentlyBuried, type: :string, example: '1', enum: %w[1 2 3]
+              property :sendingCode, type: :string, example: 'abc' # TODO: not in schema.  remove?
+              property :currentlyBuriedPersons, type: :array, description: 'data about uploaded attachments' do
```
Comment (jholton) [inline app/swagger/requests/preneeds_claims.rb]:
```suggestion
              property :currentlyBuriedPersons, type: :array, description: 'data about persons currently buried under the veteran's eligibility' do
```

**Comment by cehsu** [comment]
@lihanli this is ready for another look when convenient.

**Pull request merged**